src/librados/librados.cc

```diff
@@ -1032,11 +1036,22 @@ uint64_t librados::IoCtx::pool_required_alignment()
   return io_ctx_impl->client->pool_required_alignment(get_id());
 }
 
+std::string& librados::IoCtx::get_pool_name() const
+{
+  std::string pn;
```

Comment:
should also add a lock for the cached_pool_names list to IoCtxImpl, and hold it throughout this function, so it's safe to call concurrently